packages/compiler/src/lib/decorators.ts

```diff
@@ -417,12 +419,20 @@ export function $pattern(
   }
 
   context.program.stateMap(patternValuesKey).set(target, pattern);
+
+  if (validationMessage) {
+    context.program.stateMap(patternValidationMessageKey).set(target, validationMessage);
```

Comment:
Why not change the type and use the existing state key

Reply:
Good call, I like the idea of doing getPatternData and having it return an object that we can extend in future.